pkgs/development/php-packages/box/default.nix

```diff
@@ -1,28 +1,20 @@
-{ mkDerivation, fetchurl, makeWrapper, lib, php }:
+{ fetchFromGitHub, lib, php }:
 
-let
+php.buildPhpProject (finalAttrs: {
   pname = "box";
-  version = "4.2.0";
-in
-mkDerivation {
-  inherit pname version;
+  version = "4.3.8";
 
-  src = fetchurl {
-    url = "https://github.com/box-project/box/releases/download/${version}/box.phar";
-    sha256 = "sha256-9pPhqFq9t3MKI/y6/7iCYB8ddWFrafGVcV/k+Exb+KQ=";
+  src = fetchFromGitHub {
+    owner = "box-project";
+    repo = "box";
+    rev = finalAttrs.version;
+    hash = "sha256-v1J84nqaX36DrLLH5kld+8NIymqtt5/5nJWJNCBVFRE=";
   };
 
-  dontUnpack = true;
+  vendorHash = "sha256-cA9nuB+72N2tpYrojANOgYslOMWYeNe/+F41q211rTk=";
 
-  nativeBuildInputs = [ makeWrapper ];
-
-  installPhase = ''
-    runHook preInstall
-    mkdir -p $out/bin
-    install -D $src $out/libexec/box/box.phar
-    makeWrapper ${php}/bin/php $out/bin/box \
-      --add-flags "-d phar.readonly=0 $out/libexec/box/box.phar"
```

Comment:
IIRC, tnis is needed for box to be useable for creating phar archives.

Reply:
Since we are using the source distribution, I don't think it's needed.

Reply:
Box is a tool that creates phar archives. Without this option PHP will not allow creating phar archives at runtime.

Reply:
Ok I will double check.

Reply:
Fixed now.